hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/utils/BackgroundService.java

```diff
@@ -150,6 +180,18 @@ public void run() {
         execTaskCompletion();
       }
 
+      if (schedulingMode.shouldSleepAfterCompletion()) {
+        try {
+          long delayMillis = getIntervalMillis();
+          LOG.debug("Waiting {} ms after task completion before next execution", delayMillis);
+          Thread.sleep(delayMillis);
+        } catch (InterruptedException e) {
+          LOG.warn("Interrupted while waiting for fixed interval");
+          execTaskCompletion();
```

Comment:
execTaskCompletion() is called after future.join().  I think we don't need to call it here again.